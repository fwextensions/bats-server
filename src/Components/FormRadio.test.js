import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import FormRadio from './FormRadio';
import FormRadioFieldSet from './FormRadioFieldSet';

test('Snapshot', () => {
  const component = renderer.create(
    <FormRadioFieldSet
      labelText="Select Favorite Fruit"
      property="fruit"
      isRequired
      validationState='FIXED'
    >
      <FormRadio
        onChange={() => {}}
        currentValue="APPLE"
        label="Apple"
        property="fruit"
        value="APPLE"
      />
      <FormRadio
        onChange={() => {}}
        currentValue="APPLE"
        label="Banana"
        property="fruit"
        value="BANANA"
      />
      <FormRadio
        onChange={() => {}}
        currentValue="APPLE"
        label="Coconut"
        property="fruit"
        value="COCONUT"
      />
    </FormRadioFieldSet>
  );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});