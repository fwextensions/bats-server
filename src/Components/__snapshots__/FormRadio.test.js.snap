// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<fieldset
  className="usa-fieldset form-radio-field-set form-radio-field-set--success"
>
  <div
    className="form-radio-field-set__background"
  >
    <label
      className="usa-label usa-label--required"
      htmlFor="fruit"
    >
      Select Favorite Fruit
    </label>
    <div
      className="usa-radio"
    >
      <input
        checked={true}
        className="usa-radio__input"
        disabled={false}
        id="fruit-APPLE"
        name="fruit"
        onChange={[Function]}
        type="radio"
        value="APPLE"
      />
      <label
        className="usa-radio__label"
        htmlFor="fruit-APPLE"
      >
        Apple
      </label>
    </div>
    <div
      className="usa-radio"
    >
      <input
        checked={false}
        className="usa-radio__input"
        disabled={false}
        id="fruit-BANANA"
        name="fruit"
        onChange={[Function]}
        type="radio"
        value="BANANA"
      />
      <label
        className="usa-radio__label"
        htmlFor="fruit-BANANA"
      >
        Banana
      </label>
    </div>
    <div
      className="usa-radio"
    >
      <input
        checked={false}
        className="usa-radio__input"
        disabled={false}
        id="fruit-COCONUT"
        name="fruit"
        onChange={[Function]}
        type="radio"
        value="COCONUT"
      />
      <label
        className="usa-radio__label"
        htmlFor="fruit-COCONUT"
      >
        Coconut
      </label>
    </div>
    <div
      className="usa-success-message margin-top-1"
    >
      <i
        className="fas fa-check-circle"
      />
       Success
    </div>
  </div>
</fieldset>
`;
